icu4c/source/i18n/unicode/messageformat2_formattable.h

```diff
@@ -64,6 +65,40 @@ namespace message2 {
         virtual ~FormattableObject();
     }; // class FormattableObject
 
+    /**
+     * The `DateInfo` struct represents all the information needed to
+     * format a date with a time zone. It includes an absolute date and a time zone name,
+     * as well as a calendar name. The calendar name is not currently used.
+     *
+     * @internal ICU 76 technology preview
+     * @deprecated This API is for technology preview only.
+     */
+    struct U_I18N_API DateInfo {
+        /**
+         * Milliseconds since Unix epoch
+         *
+         * @internal ICU 76 technology preview
+         * @deprecated This API is for technology preview only.
+         */
+        UDate date;
+        /**
+         * IANA time zone name; "UTC" if UTC; empty string if value is floating
+         *
+         * @internal ICU 76 technology preview
+         * @deprecated This API is for technology preview only.
+         */
+        UnicodeString zoneName;
+        /**
+         * Calendar name. Note: non-Gregorian calendars not yet implemented.
+         * String is empty if calendar not specified. Proleptic Gregorian calendar
+         * is default.
+         *
+         * @internal ICU 76 technology preview
+         * @deprecated This API is for technology preview only.
+         */
+        UnicodeString calendarName;
```

Comment:
What is the calendarName supposed to map to? Is it the 'old style name' (`gregorian`) matching https://github.com/unicode-org/icu/blob/cf7ff1b0a508c216c57b61badfa44681cf0978fc/icu4c/source/i18n/unicode/calendar.h#L1297 or a bcp47 `-u-ca` type (`gregory`)? 

If it's from the locale, wouldn't that be visible from the *locale* of the message format itself?

Reply:
The spec currently says that it's [this](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeCalendarIdentifier), but it's not fully defined in v45 (the calendar option is reserved for future standardization). Probably we want to support whatever JS Temporal does or at least be compatible with that.

Reply:
It should probably link there, as a 'calendar type' in ICU4C is one of the old types generally.

Reply:
@srl295 Added a comment in [82f9ddc](https://github.com/unicode-org/icu/pull/3012/commits/82f9ddc591347bc3805230dbca8c9cea25bd4191) - let me know if that looks OK.